zilliqa/src/consensus.rs

```diff
@@ -1947,20 +1948,17 @@ impl Consensus {
         Ok(current.view() == 0 || current.hash() == ancestor.hash())
     }
 
-    fn check_safe_block(&mut self, proposal: &Block, during_sync: bool) -> Result<bool> {
+    fn check_safe_block(&mut self, proposal: &Block) -> Result<bool> {
         let Some(qc_block) = self.get_block(&proposal.parent_hash())? else {
             trace!("could not get qc for block: {}", proposal.parent_hash());
             return Ok(false);
         };
-        // We don't vote on blocks older than our view
-        let outdated = proposal.view() < self.view.get_view();
```

Comment:
@DrZoltanFazekas Could you double-check these changes? I think they make sense but you are the expert :)

Reply:
At [https://github.com/Zilliqa/zq2/blob/main/zilliqa/src/consensus.rs#L739-L742](https://github.com/Zilliqa/zq2/blob/main/zilliqa/src/consensus.rs#L739-L742) we would vote for the outdated proposal and we shouldn't. Most likely we would send the `Vote` to the wrong leader as it's the leader of `self.view.get_view() + 1` but if it happens to be the same leader to whom we sent a `NewView` message when we timed out in `proposal.view()`, we get caught equivocating.

Reply:
thanks, it is a little tricky to follow. The aim here is to simply move the outdated check outside of the `check_safe_block()` fn